### PR TITLE
Tabs Component initial

### PR DESCRIPTION
**Pull request by daigof**
- Similar to Form component this is the initial version compatible with PocketDerm. However:
- I dont like how it handle the Tabs as object {id,title}, worse it couples the initialTab to Id=1 forcing us to provide at least one Tab object with Id=1
- I propose drop the tab objects and just work with arrays of string titles and the active Id would be the index now. As usual this would require more refactor in PocketDerm as well.

### CODE REVIEW

**Comment by benkolde** [comment]
@daigof 

You can totally refuse this but on mobile, tabs become left aligned with the first string having a margin of `24px` from the left side of the window. It would be nice to bake this into the component if possible.

**Comment by daigof** [comment]
> You can totally refuse this but on mobile, tabs become left aligned with the first string having a margin of `24px` from the left side of the window. It would be nice to bake this into the component if possible.

@benkolde Yeah absolutely, I refactor all the styles a little bit. Note that the Tab Item already had a padding left/right: 24px so I didnt add the extra margin-left,  means that the Tab first letter will be exactly 24px to the left of the Tabs component container

File: docs/tabs.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    return <Content bgColor={tab.bgColor}>{tab.text}</Content>;
+  };
+
+  render() {
+    const { activeTabId } = this.state;
+
+    return (
+      <section>
+        <Tabs
+          activeTabId={activeTabId}
```
Comment (snags88) [inline docs/tabs.md]: Should this be `initialActiveTabId`?

File: docs/tabs.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  };
+
+  tabItems = [
+    { id: 1, text: 'Tab 1', bgColor: COLORS.postcardGreen },
+    { id: 2, text: 'Tab 2', bgColor: COLORS.postcardPeach },
+    { id: 3, text: 'Tab 3', bgColor: COLORS.postcardYellow },
+  ];
+
+  onClickTab = tab => {
+    this.setState({ activeTabId: tab.id });
```
Comment (snags88) [inline docs/tabs.md]: it looks like from the implementation, the `<Tabs />` component handles the state for which tab is currently active. Having this in the example is a bit misleading. Maybe we can just console.log out the `tab` variable and remove having `activeTabId` as state on this component

File: src/shared-components/tabs/test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import Tabs from './index';
+
+describe('<Tabs />', () => {
+  describe('default props', () => {
+    test('renders the correct props', () => {
+      const wrapper = shallow(
+        <Tabs
+          activeTabId={1}
+          tabItems={[
+            { id: 1, text: 'Rain Drop' },
```
Comment (snags88) [inline src/shared-components/tabs/test.js]: lol we can probably update the text to be more normal 😆

File: docs/tabs.md (hunk 1 above)
Comment (daigof) [reply docs/tabs.md]: lol yeah havent noticed that bug

File: docs/tabs.md (hunk 2 above)
Comment (daigof) [reply docs/tabs.md]: done

File: src/shared-components/tabs/test.js (hunk 3 above)
Comment (daigof) [reply src/shared-components/tabs/test.js]: done

**Comment by daigof** [comment]
@snags88 what do you think regarding flattening the tabItems structure to be just an array of strings (titles) and handle tabId as array index. Having refactored the Tabs Example I realized the tabItems object structure may be due that the objects contains part of the content displayed in page but I still think this should be decoupled.

**Comment by snags88** [comment]
@daigof i think we can keep as objects for now. I briefly looked at how we use it for pocketderm and it seems like we use the object notation to help with setting the query string.

**Pull request merged**